### PR TITLE
AP_HAL_ChibiOS: add STM32F7 I2C support

### PR DESCRIPTION
**Pull request by mirkix**
Add STM32F7 I2C support to AP_HAL_ChibiOS.
* Tested with NUCLEO-F767ZI (STM32F767ZI) board and HMC5983.

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/I2CDevice.cpp [hunk 1]
```diff
@@ -45,6 +45,10 @@ I2CBus I2CDeviceManager::businfo[ARRAY_SIZE_SIMPLE(I2CD)];
 #define HAL_I2C_MAX_CLOCK 100000
 #endif
 
+// values calculated with STM32CubeMX tool, PCLK=54MHz
```
Comment (magicrub) [inline libraries/AP_HAL_ChibiOS/I2CDevice.cpp]: requires a hard-coded PCLK?

**Comment by magicrub** [comment]
This logic belongs in the hwdef parsing scripts so it can generate the correct values regardless of clock settings.

**Comment by mirkix** [comment]
@magicrub Hi Tom, yes that would be nice, I agree. But PCLK should be 54MHz (max PCLK for I2C) when using a STM32F7. It's not so easy to calculate this value. You have to calculate five values (PRESC, SCLL, SCLH, SDADEL and SCLDEL) to get the result that is "hard coded" in this PR, because even in the STM32F7 reference manual there is no real formula for all these values. In the reference manual is a note "In order to get more accurate configuration values, the STM32CubeMX tool (I2C Configuration Window) should be used."

**Comment by magicrub** [comment]
lets ping @tridge and @bugobliterator

**Comment by tridge** [comment]
I think the hard-coded clock is OK for now, at least until we run across a F7 that this doesn't work for. 
In a future change I'd like to move the F4/F7 specific code into helper files, perhaps hwdef/common/stm32_f4/*.c and hwdef/common/stm32_f7/*.c

**Comment by mirkix** [comment]
@tridge You are right, fixed with some small improvement.

**Comment by tridge** [comment]
tested on a F427

**Pull request merged**